resources/qml/Settings/SettingExtruder.qml

```diff
@@ -72,6 +72,24 @@ SettingItem
             value: control.currentText != "" ? control.model.getItem(control.currentIndex).color : ""
         }
 
+        Binding
+        {
+            target: control
+            property: "currentIndex"
+            value:
+            {
+                if(propertyProvider.properties.value == -1)
+                {
+                    // TODO: accidently the extruder position is also the index. fix it
+                    return Cura.MachineManager.defaultExtruderPosition;
```

Comment:
It's a bit hard to predict, but I think that if you set the extruder setting manually to the default (usually 0) this will cause the setting to be marked as changed even though it hasn't. The reset arrow will appear but won't do anything. The profile will be marked as containing user changes.

Reply:
It will mark it as changed, but I don't think that's a problem.
If the user explicitly sets the extruder, then IMO it can be regarded as a user setting and it will not change upon disabling / enabling extruders (unless that specific extruder has been turned off, then the setting is removed)